libs/clients/contentful-graphql/src/lib/cache.ts

```diff
@@ -0,0 +1,41 @@
+import { ConfigType } from '@island.is/nest/config'
+import { createRedisCacheManager } from '@island.is/cache'
+import {
+  buildCacheControl,
+  CacheConfig,
+  Request,
+} from '@island.is/clients/middlewares'
+
+import { ContentfulGraphQLClientConfig } from './contentful-graphql.config'
+
+
+function overrideCacheControl(request: Request) {
+  return buildCacheControl({ 
+    maxAge: 10 * 60, // 10 minutes
+    public: true // required to enable caching for this contentful gql
+  })
+}
+
+
+export const getCache = async (
+  config: ConfigType<typeof ContentfulGraphQLClientConfig>,
+): Promise<CacheConfig | undefined> => {
+  if (config.redis.nodes.length === 0) {
+    console.warn('No redis nodes defined, cache will not be used')
+    return undefined
+  }
+  const cacheManager = await createRedisCacheManager({
+    name: 'clients-contentful-graphql',
+    nodes: config.redis.nodes,
+    ssl: config.redis.ssl,
+    noPrefix: true,
+    ttl: 0
+  })
+
+  return {
+    cacheManager,
+    cacheKey: (request: Request) => request.url + JSON.stringify(request.body),
```

Comment:
If the library is only making GET request the url is unique, and you can rely on the default function.